toolkit/app-gen-toc.py

```diff
@@ -685,7 +685,7 @@ def main():
     parser.add_argument(
         "--config-dir",
         type=str,
-        default="",
+        default=Path(os.path.dirname(__file__)) / "build/config",
```

Comment:
Perhaps instead of this, if the config file is not found, search for it in `build/config`? This way the Makefile could still pass a custom config dir, like the board dir, that contains no files.

Reply:
I'd rather not have any auto-things, it's too easy for things to go wrong, eg accidentally use the wrong file and you don't realise.

Having a default like you've got it is good.

Reply:
Note though that if you do want to customize a config, you'll have to pass its directory in the Makefile somehow to `--config-dir`.